app/mixins/donation-route.js

```diff
@@ -0,0 +1,22 @@
+import Ember from 'ember';
+
+const {
+  get,
+  inject: { service },
+  Mixin
+} = Ember;
+
+const ALREADY_A_SUBSCRIBER = "You're already supporting this project.";
+
+export default Mixin.create({
+  flashMessages: service(),
+
+  afterModel({ project, subscription }) {
+    if (subscription) {
+      get(this, 'flashMessages').success(ALREADY_A_SUBSCRIBER);
+      this.transitionTo('project', project);
+    } else {
+      this._super.call(...arguments);
+    }
+  }
+});
```

Comment:
This mixin works, and is probably the next best thing, but combined with the fact that these routes have an explicitly separate layout to other project routes, it's really further telling us that our onboarding routes ought to be grouped under a parent route of some sort. That would allow us to have this `afterModel` as well as our `model` within that parent route. 

It's not something suitable for this route, but it is something to consider and probably discuss.